### PR TITLE
update kubectl zh-cn

### PR DESCRIPTION
**Pull request by hwdef**
#### What type of PR is this?

/kind documentation

#### What this PR does / why we need it:

Translate Chinese in kubectl i18n. This PR only updates the `kubectl -h` section

**process**
1. `./hack/update-translations.sh -x ` get pot file
2.  open zh-cn po file by poedit, import pot file in step 1
3. translate

before:

```shell
$ ./_output/bin/kubectl --help 
kubectl controls the Kubernetes cluster manager.

 Find more information at: https://kubernetes.io/docs/reference/kubectl/

Basic Commands (Beginner):
  create          通过文件名或者标准输入流(stdin)创建一个资源
  expose          使用 replication controller, service, deployment 或者 pod 并暴露它作为一个 新的 Kubernetes
Service
  run             在集群上运行特定镜像
  set             为对象设置指定特性

Basic Commands (Intermediate):
  explain         查看资源的文档
  get             显示一个或多个资源
  edit            编辑服务器上的资源
  delete          Delete resources by filenames, stdin, resources and names, or by resources and
label selector

Deploy Commands:
  rollout         更新某资源上的标签
  scale           为 Deployment, ReplicaSet, Replication Controller 或者 Job 设置一个新的副本数量
  autoscale       自动调整一个 Deployment, ReplicaSet, 或者 ReplicationController 的副本数量

Cluster Management Commands:
  certificate     修改证书资源。
  cluster-info    显示集群信息
  top             显示 Resource (CPU/Memory) 使用.
  cordon          标记节点为不可调度
  uncordon        标记节点为可调度
  drain           清空节点以准备维护
  taint           更新一个或者多个节点上的污点

Troubleshooting and Debugging Commands:
  describe        显示特定资源或资源组的详细信息
  logs            打印 Pod 中容器的日志
  attach          挂接到一个运行中的容器
  exec            在某个容器中执行一个命令
  port-forward    将一个或多个本地端口转发到某个 Pod
  proxy           运行一个指向 Kubernetes API 服务器的代理
  cp              将文件和目录复制到容器中或从容器中复制出来。
  auth            Inspect authorization
  debug           Create debugging sessions for troubleshooting workloads and nodes
  events          List events

Advanced Commands:
  diff            Diff the live version against a would-be applied version
  apply           通过文件名或标准输入流(stdin)对资源进行配置
  patch           更新某资源上的标签
  replace         通过 filename 或者 stdin替换一个资源
  wait            Experimental: Wait for a specific condition on one or many resources
  kustomize       Build a kustomization target from a directory or URL

Settings Commands:
  label           更新某资源上的标签
  annotate        更新一个资源的注解
  completion      为指定的 Shell(Bash 或 zsh) 输出 Shell 补全代码。

Other Commands:
  api-resources   编辑服务器上的资源
  api-versions    
                # 输出支持的 API 版本
                kubectl api-versions
  config          修改 kubeconfig 文件
  plugin          Provides utilities for interacting with plugins
  version         输出客户端和服务端的版本信息

Usage:
  kubectl [flags] [options]

Use "kubectl <command> --help" for more information about a given command.
Use "kubectl options" for a list of global command-line options (applies to all commands).
```

after:

```shell
$ ./_output/bin/kubectl --help 
kubectl controls the Kubernetes cluster manager.

 Find more information at: https://kubernetes.io/docs/reference/kubectl/

Basic Commands (Beginner):
  create          通过文件名或者标准输入流(stdin)创建一个资源
  expose          将 replication controller, service, deployment 或者 pod 暴露为一个新的 Kubernetes service
  run             在集群上运行特定镜像
  set             为对象设置指定特性

Basic Commands (Intermediate):
  explain         查看资源的文档
  get             显示一个或多个资源
  edit            编辑服务端上的资源
  delete          按文件名、stdin、资源和名称或资源和标签选择器删除资源

Deploy Commands:
  rollout         管理资源的上线
  scale           为 deployment, replicaSet, replication controller 设置一个新的副本数量
  autoscale       自动调整一个 deployment, replica set, stateful set 或者 replication controller 的副本数量

Cluster Management Commands:
  certificate     修改证书资源。
  cluster-info    显示集群信息
  top             显示资源 (CPU/Memory) 使用量
  cordon          标记节点为不可调度
  uncordon        标记节点为可调度
  drain           清空节点以准备维护
  taint           更新一个或者多个节点上的污点

Troubleshooting and Debugging Commands:
  describe        显示特定资源或资源组的详细信息
  logs            打印 Pod 中容器的日志
  attach          连接到正在运行的容器
  exec            在容器中执行一个命令
  port-forward    将一个或多个本地端口转发到 pod
  proxy           运行一个指向 Kubernetes API 服务器的代理
  cp              将文件和目录复制到容器中或从容器中复制出来
  auth            Inspect authorization
  debug           创建调试会话，以排除工作负载和节点的故障
  events          列出事件

Advanced Commands:
  diff            将实时版本与可能应用的版本进行比较
  apply           通过文件名或标准输入流(stdin)对资源进行配置
  patch           更新资源的字段
  replace         通过文件名或者 stdin替换一个资源
  wait            实验性：在一个或多个资源上等待特定条件的出现
  kustomize       Build a kustomization target from a directory or URL

Settings Commands:
  label           更新资源上的标签
  annotate        更新一个资源的注解
  completion      为指定的 shell(bash, zsh, fish, or powershell) 输出 shell 补全代码。

Other Commands:
  api-resources   打印服务端上支持的 API 资源
  api-versions    以“组/版本”的形式打印服务端支持的 API 版本
  config          修改 kubeconfig 文件
  plugin          提供与插件交互的实用程序
  version         输出客户端和服务端的版本信息

Usage:
  kubectl [flags] [options]

Use "kubectl <command> --help" for more information about a given command.
Use "kubectl options" for a list of global command-line options (applies to all commands).
```

#### Which issue(s) this PR fixes:

Fixes #

#### Special notes for your reviewer:

The reason this PR seems to have a lot of line changes is because I imported the generated pot file (you can check the `process` steps above), and I only translated what `kubectl -h` involves, and the rest of it I can assure you that I didn't change, so I think you can just REVIEW the part that involves `kubectl -h`, which will reduce a lot of the workload.

And I can guarantee that I will change all the kubectl translations before the next kubernetes(1.29) release.

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by hwdef** [comment]
@tengqm 
Please REVIEW this PR if you have time!

**Comment by hwdef** [comment]
@atiratree 
Did I do it in the right order?
```
1. ./hack/update-translations.sh -x  get pot file
2. open zh-cn po file by poedit, import pot file in step 1
3. translate
```
I'd also like to ask under what circumstances should `-k` and `-g` be used? What order should `-k -g -x` be used?

File: staging/src/k8s.io/kubectl/pkg/util/i18n/translations/kubectl/zh_CN/LC_MESSAGES/k8s.po [hunk 1]
```diff
@@ -8,19 +8,94 @@ msgid ""
...
+#, fuzzy
+msgid ""
+"\n"
+"\t\t  # Create a cluster role binding for user1, user2, and group1 using the "
+"cluster-admin cluster role\n"
+"\t\t  kubectl create clusterrolebinding cluster-admin --clusterrole=cluster-"
+"admin --user=user1 --user=user2 --group=group1"
+msgstr ""
+"\n"
+"\t\t  # 使用 cluster-admin ClusterRole 为 user1, user2, and group1 创建一个 "
```
Comment (windsonsea) [inline staging/src/k8s.io/kubectl/pkg/util/i18n/translations/kubectl/zh_CN/LC_MESSAGES/k8s.po:33]:
```suggestion
"\t\t  # 使用 cluster-admin ClusterRole 为 user1、user2 和 group1 创建一个 "
```

File: staging/src/k8s.io/kubectl/pkg/util/i18n/translations/kubectl/zh_CN/LC_MESSAGES/k8s.po [hunk 2]
```diff
@@ -8,19 +8,94 @@ msgid ""
...
+"\n"
+"\t\t  # Create a new config map named my-config from an env file\n"
+"\t\t  kubectl create configmap my-config --from-env-file=path/to/foo.env --"
+"from-env-file=path/to/bar.env"
+msgstr ""
+"\n"
+"\t\t  # 通过文件夹 bar 创建一个名称为 my-config 的 configmap\n"
+"\t\t  kubectl create configmap my-config --from-file=path/to/bar\n"
+"\n"
+"\t\t  # 创建一个名称为 my-config 的 configmap 并指定 keys 而不是使用磁盘上所"
```
Comment (windsonsea) [inline staging/src/k8s.io/kubectl/pkg/util/i18n/translations/kubectl/zh_CN/LC_MESSAGES/k8s.po:67]:
```suggestion
"\t\t  # 创建一个名称为 my-config 的 configmap 并指定 key 而不是使用磁盘上所"
```

File: staging/src/k8s.io/kubectl/pkg/util/i18n/translations/kubectl/zh_CN/LC_MESSAGES/k8s.po [hunk 3]
```diff
@@ -8,19 +8,94 @@ msgid ""
...
+"\t\t  kubectl create secret docker-registry my-secret --docker-"
+"server=DOCKER_REGISTRY_SERVER --docker-username=DOCKER_USER --docker-"
+"password=DOCKER_PASSWORD --docker-email=DOCKER_EMAIL\n"
+"\n"
+"\t\t  # Create a new secret named my-secret from ~/.docker/config.json\n"
+"\t\t  kubectl create secret docker-registry my-secret --from-file=."
+"dockerconfigjson=path/to/.docker/config.json"
+msgstr ""
+"\n"
+"\t\t  # 如果你还没有 .dockercfg 文件, 你可以直接使用下面的命令创建一个 "
```
Comment (windsonsea) [inline staging/src/k8s.io/kubectl/pkg/util/i18n/translations/kubectl/zh_CN/LC_MESSAGES/k8s.po:92]:
```suggestion
"\t\t  # 如果你还没有 .dockercfg 文件，你可以直接使用下面的命令创建一个 "
```
应该可以用中文标点吧？

File: staging/src/k8s.io/kubectl/pkg/util/i18n/translations/kubectl/zh_CN/LC_MESSAGES/k8s.po [hunk 4]
```diff
@@ -36,274 +111,2498 @@ msgstr ""
...
+"\t\tkubectl cp <some-namespace>/<some-pod>:/tmp/foo /tmp/bar"
 msgstr ""
 "\n"
-"\t\t# 输出所有命令继承的 flags\n"
-"\t\tkubectl options"
+"\t    # !!!注意!!!\n"
+"\t    # 要求容器中有 'tar' 命令\n"
+"\t    # image.  If 'tar' is not present, 'kubectl cp' will fail.\n"
+"\n"
+"\t    # 复制本地目录 /tmp/foo_dir 到 default namespace 下的远程 pod 的 /tmp/"
```
Comment (windsonsea) [inline staging/src/k8s.io/kubectl/pkg/util/i18n/translations/kubectl/zh_CN/LC_MESSAGES/k8s.po:154]:
```suggestion
"\t    # 复制本地目录 /tmp/foo_dir 到 default namespace 下的远程 Pod 的 /tmp/"
```
Pod 写法要跟 [Pod Concept](https://kubernetes.io/zh-cn/docs/concepts/workloads/pods/) 一致吗

File: staging/src/k8s.io/kubectl/pkg/util/i18n/translations/kubectl/zh_CN/LC_MESSAGES/k8s.po [hunk 5]
```diff
@@ -36,274 +111,2498 @@ msgstr ""
...
 "\n"
-"\t\t# 显示指定 Pod 和它的容器的 metrics\n"
-"\t\tkubectl top pod POD_NAME --containers\n"
+"\t\t# 将 ‘pod.yaml' 指定的资源的实时状态转换为最新版本\n"
+"\t\t# 并以 json 格式打印到 stdout.\n"
+"\t\tkubectl convert -f pod.yaml --local -o json\n"
 "\n"
-"\t\t# 显示指定 label 为 name=myLabel 的 Pods 的 metrics\n"
-"\t\tkubectl top pod -l name=myLabel"
+"\t\t# 将当前目录下的所以文件转换为最新版本并创建它们.\n"
```
Comment (windsonsea) [inline staging/src/k8s.io/kubectl/pkg/util/i18n/translations/kubectl/zh_CN/LC_MESSAGES/k8s.po:259]:
```suggestion
"\t\t# 将当前目录下的所有文件转换为最新版本并创建它们。\n"
```

**Comment by hwdef** [comment]
@windsonsea 
Thanks for your review, But I haven't translated these yet, so please refrain from REVIEWING them for now, please refer to
image

**Comment by atiratree** [comment]
> I'd also like to ask under what circumstances should -k and -g be used? What order should -k -g -x be used?

@hwdef  basically what it says and in order it does in help

```
 -x extract strings to a POT file
 -k fix .po files; deprecate translations by marking them obsolete and supply default messages
 -g sort .po files and generate .mo files
```

keep in mind that the -k might mark obsolete any change (e.g. punctuation in the sentence), and the old translation is very likely to be reusable (with minor changes)

**Comment by hwdef** [comment]
@atiratree 
I see, thank you!

**Pull request closed without merging**